app/org/sagebionetworks/bridge/play/controllers/TaskController.java

```diff
@@ -30,14 +34,28 @@ public void setTaskService(TaskService taskService) {
         this.taskService = taskService;
     }
     
-    public Result getTasks(String untilString) throws Exception {
+    public Result getTasks(String untilString, String offset, String daysAhead) throws Exception {
         UserSession session = getAuthenticatedAndConsentedSession();
         
-        DateTime endsOn = (untilString == null) ? 
-            DateTime.now().plusDays(TaskService.DEFAULT_EXPIRES_ON_DAYS) :
-            DateTime.parse(untilString);
+        DateTime endsOn = null;
+        DateTimeZone zone = null;
         
-        List<Task> tasks = taskService.getTasks(session.getUser(), endsOn);
+        if (StringUtils.isNotBlank(untilString)) {
+            // Old API, infer time zone from the until parameter. This is not ideal.
+            endsOn = DateTime.parse(untilString);
+            zone = endsOn.getZone();
+        } else if (StringUtils.isNotBlank(daysAhead) && StringUtils.isNotBlank(offset)) {
+            zone = DateUtils.parseZoneFromOffsetString(offset);
+            int numDays = Integer.parseInt(daysAhead);
+            // When querying for days, we ignore the time of day of the request and query to then end of the day.
+            endsOn = DateTime.now(zone).plusDays(numDays)
+                .withHourOfDay(23).withMinuteOfHour(59).withSecondOfMinute(59);
+        } else {
+            throw new BadRequestException("Supply either 'until' parameter, or 'daysAhead' and 'offset' parameters.");
+        }
```

Comment:
It's parsing HTTP parameters, so I feel like it belongs in the controller. I tried partially refactoring this to ScheduleContext as a parameters object, but it's still messy. Will keep thinking about this.